### ISSUE TITLE
Version 7.0.1: Getting ConnectionError(a float is required) when passing request_timeout in the client calls

### ISSUE THREAD

**Issue by jyothikp** [elastic/elasticsearch-py#969]
I am using the version 7.0.1 and getting this error when passing the request_timeout in client calls:
es.indices.delete_template(
                name=template_name, 
                master_timeout="60s",
                request_timeout= 90
            )
`Traceback (most recent call last):
    \n  File "<ipython-input-13-d6617379fbe8>", line 4, in <module>    \n    request_timeout=template_timeouts.get(\'request_timeout_secs\', 90)    \n  File "build/bdist.linux-x86_64/egg/elasticsearch/client/utils.py", line 84, in _wrapped    \n    return func(*args, params=params, **kwargs)    \n  File "build/bdist.linux-x86_64/egg/elasticsearch/client/indices.py", line 607, in delete_template    \n    "DELETE", _make_path("_template", name), params=params    \n  File "build/bdist.linux-x86_64/egg/elasticsearch/transport.py", line 353, in perform_request    \n    timeout=timeout,    \n  File "build/bdist.linux-x86_64/egg/elasticsearch/connection/http_urllib3.py", line 232, in perform_request    \n    raise ConnectionError("N/A", str(e), e)    \nConnectionError: ConnectionError(a float is required) caused by: TypeError(a float is required)    \n`

I am making a delete_template call here. I am suspecting it started coming after this change:
https://github.com/elastic/elasticsearch-py/pull/937, as it was working in version 7.0.0. With this change request_timeout will go as normal query_params and _escape will be called on this value which will convert it to a string. While urllib library is expecting this parameter as a float value. Please let me know if any extra information required around the issue.

**Comment by tj8000rpm** [comment]
I have a same issue at ``update_aliases``. It was working in version 6.3.

Here is my code and my stack traces.

**code**
```python
from elasticsearch import Elasticsearch
es = Elasticsearch('172.17.0.1:9200')

update_alias={
    "actions" : [
        { "remove" : { "index" : "hoge1", "alias" : "hoge" } },
        { "add" :    { "index" : "hoge2", "alias" : "hoge" } }
    ]
}
es.indices.update_aliases(body=update_alias, request_timeout=30)
```

**traceback**
```
Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/timeout.py", line 130, in _validate_timeout
    if value <= 0:
TypeError: '<=' not supported between instances of 'str' and 'int'

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 232, in perform_request
    method, url, body, retries=Retry(False), headers=request_headers, **kw
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 590, in urlopen
    timeout_obj = self._get_timeout(timeout)
  File "/usr/local/lib/python3.7/site-packages/urllib3/connectionpool.py", line 301, in _get_timeout
    return Timeout.from_float(timeout)
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/timeout.py", line 155, in from_float
    return Timeout(read=timeout, connect=timeout)
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/timeout.py", line 94, in __init__
    self._connect = self._validate_timeout(connect, 'connect')
  File "/usr/local/lib/python3.7/site-packages/urllib3/util/timeout.py", line 137, in _validate_timeout
    "int, float or None." % (name, value))
ValueError: Timeout value connect was 30, but it must be an int, float or None.

During handling of the above exception, another exception occurred:

Traceback (most recent call last):
  File "hoge.py", line 40, in <module>
    es.indices.update_aliases(body=update_alias, request_timeout=30)
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/client/utils.py", line 84, in _wrapped
    return func(*args, params=params, **kwargs)
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/client/indices.py", line 495, in update_aliases
    "POST", "/_aliases", params=params, body=body
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/transport.py", line 353, in perform_request
    timeout=timeout,
  File "/usr/local/lib/python3.7/site-packages/elasticsearch/connection/http_urllib3.py", line 244, in perform_request
    raise ConnectionError("N/A", str(e), e)
elasticsearch.exceptions.ConnectionError: ConnectionError(Timeout value connect was 30, but it must be an int, float or None.) caused by: ValueError(Timeout value connect was 30, but it must be an int, float or None.)
```

**Comment by tj8000rpm** [comment]
I send a PR(#976).

**Comment by Winterflower** [comment]
Thank you for reporting this issue @jyothikp and thank you for the details to reproduce @tj8000rpm ! I will try to reproduce this locally so I can confirm what has caused this and will report back with my findings.

**Comment by nsonic001** [comment]
@Winterflower still facing this issue with 7.0.5 even for 
`es.indices.put_template(name=name, body=body, master_timeout="120s", request_timeout=120)`
this request.

```
ConnectionError                           Traceback (most recent call last)
<ipython-input-28-49d21007cf85> in <module>()
----> 1 es.indices.put_template(name=name, body=body, master_timeout="120s", request_timeout=float(120.0))

/usr/local/lib/python2.7/site-packages/elasticsearch/client/utils.pyc in _wrapped(*args, **kwargs)
     82                 if p in kwargs:
     83                     params[p] = kwargs.pop(p)
---> 84             return func(*args, params=params, **kwargs)
     85
     86         return _wrapped

/usr/local/lib/python2.7/site-packages/elasticsearch/client/indices.pyc in put_template(self, name, body, params)
    551                 raise ValueError("Empty value passed for a required argument.")
    552         return self.transport.perform_request(
--> 553             "PUT", _make_path("_template", name), params=params, body=body
    554         )
    555

/usr/local/lib/python2.7/site-packages/elasticsearch/transport.pyc in perform_request(self, method, url, headers, params, body)
    348                     headers=headers,
    349                     ignore=ignore,
--> 350                     timeout=timeout,
    351                 )
    352

/usr/local/lib/python2.7/site-packages/elasticsearch/connection/http_urllib3.pyc in perform_request(self, method, url, params, body, timeout, ignore, headers)
    248             if isinstance(e, ReadTimeoutError):
    249                 raise ConnectionTimeout("TIMEOUT", str(e), e)
--> 250             raise ConnectionError("N/A", str(e), e)
    251
    252         # raise errors based on http status codes, let the client handle those if needed

ConnectionError: ConnectionError(a float is required) caused by: TypeError(a float is required)
```

**Comment by nsonic001** [comment]
@tj8000rpm I don't know if you noticed this or not, but while we execute `put_template` and `delete_template` with `request_timeout` It actually does the operation but response is a failure. 
basically, If I execute `es.indices.put_template(name=name, body=body, master_timeout="120s", request_timeout=120)` it throws exception of `ConnectionError` but If I check the templates in es. template are getting added/updated. This is weird.
@Winterflower just for reference.

**Comment by HonzaKral** [comment]
Fix has just been released as `7.1.0`, hope this helps!
